shadowsocks/cipher_testing.go

```diff
@@ -22,21 +22,25 @@ import (
 	"github.com/shadowsocks/go-shadowsocks2/shadowaead"
 )
 
+const testCipher = "chacha20-ietf-poly1305"
+
 // MakeTestCiphers creates a CipherList containing `numCiphers` fresh AEAD ciphers.
-func MakeTestCiphers(numCiphers int) (CipherList, error) {
+func MakeTestCiphers(numCiphers int) (CipherList, []string, error) {
```

Comment:
Could you decompose this into MakeTestSecrets and MakeTestCiphers? You end up with simpler methods, no need to conflate them. You can then call MakeTestCiphers(makeTestSecrets(n)).
Wherever you need the secrets, you can assign to a variable.

Reply:
Done